src/dnf-plugins/product-id/productdb.c

```diff
@@ -132,7 +134,8 @@ void readProductDb(ProductDb *productDb, GError **err) {
  * @param productDb populated ProductDb
  * @param err a pointer to a pointer to a glib error.  Updated if an error occurs.
  */
-void writeProductDb(ProductDb *productDb, GError **err) {
+int writeProductDb(ProductDb *productDb) {
+    int ret = 0;
```

Comment:
Returned values is used in `writeRepoMap()`. As you can see we returned the error, but we didn't use it for anything. We just printed it in this function `writeProductDb()`. We use `printError()` for printing errors in this function. This function log error and free memory of error. So it is IMHO better to print error in this function `writeProductDb()` with some context than return this error, where it is useless.

Reply:
What @jirihnidek says makes sense to me.

Reply:
OK wrt `printError()`, however my note about the `int` return value still stands: it is only checked as == or != 0, so adding custom return values that are not currently used is pointless. It can be always added later _if needed_.

Reply:
Changed the code to return `gboolean` value.